lib/membrane_http_adaptive_stream/sink.ex

```diff
@@ -241,6 +326,108 @@ defmodule Membrane.HTTPAdaptiveStream.Sink do
     end
   end
 
+  # we are operating on partial segments which may require
+  # assembling previous partial segments and creating regular one
+  # before processing the current segment
+  defp handle_partial_segment(
```

Comment:
I think that handling partial segments and any HLS-specific stuff should be done in `Membrane.HTTPAdaptiveStream.HLS`. At least it should be moved out of the sink

Reply:
`Membrane.HTTPAdaptiveStream.HLS` is just responsible for serializing HLS manifest so I can't see how the logic could be moved there.

On the other hand, the use case for the sink would also be to support DASH and with LL-DASH there is also a concept of partial segments. Well, kind of as the segments should be served with chunked transfer encoding, and since the partial segments can be just concatenated the specific storage could handle that (receiving partial segments and making sure that it transfers them on the same HTTP connection).

Reply:
The more I think about it the more it appears that we probably shouldn't try to make the `Sink` to be suitable for both HLS and DASH, at first it may seem that both of them just need different manifest serialization but in fact, any of them may need to handle their own internal state that has nothing common with the other protocol.

Reply:
> Membrane.HTTPAdaptiveStream.HLS is just responsible for serializing HLS manifest so I can't see how the logic could be moved there.

Maybe it could generate some data needed for that manifest as well, like byte ranges or time

> since the partial segments can be just concatenated the specific storage could handle that

Good idea, I'd go for it. Some common stuff could be handled by `Membrane.HTTPAdaptiveStream.Storage`.

> we probably shouldn't try to make the Sink to be suitable for both HLS and DASH

I'd say we should move as many things out of the element as possible. Then it won't really matter if we have one sink or two. And anybody can create their custom sink easily.

Reply:
> Maybe it could generate some data needed for that manifest as well, like byte ranges or time
Tried to move as much logic to hls manifest serialization in recent commit, you can look that up 😉